.github/scripts/mr_generate_summary.py

```diff
@@ -1,26 +1,22 @@
 # Collect the results of the various model test runs which are done as part of
 # the model regression CI pipeline and dump them as a single file artifact.
 # This artifact will the then be published at the end of the tests.
-import copy
+from collections import defaultdict
 import json
 import os
 from pathlib import Path
-from typing import Dict
+from typing import Dict, List
 
 
 def combine_result(
-    result1: Dict[str, dict], result2: Dict[str, dict]
-) -> Dict[str, dict]:
-    combined_dict = copy.deepcopy(result1)
-
-    for dataset, results_for_dataset in result2.items():
-        for config, res in results_for_dataset.items():
-
-            if dataset not in combined_dict:
-                combined_dict[dataset] = {}
-
-            assert config not in combined_dict[dataset]
-            combined_dict[dataset][config] = res
+    result1: Dict[str, dict], result2: Dict[str, Dict[str, Dict]]
+) -> Dict[str, Dict[str, List]]:
```

Comment:
Ik we don't require docstrings here, but maybe a comment that we expect a dict of key: dataset, and value: (dict of key: config value: list of results)